common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationResultFactory.java

```diff
@@ -93,11 +113,18 @@ public MicrosoftStsAuthorizationResult createAuthorizationResult(final int resul
                 result = createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
                         MicrosoftAuthorizationErrorResponse.DEVICE_REGISTRATION_NEEDED,
                         MicrosoftAuthorizationErrorResponse.DEVICE_REGISTRATION_NEEDED);
+                // Set username returned from the service
+                result.getAuthorizationErrorResponse().setUserName(data.getStringExtra(
```

Comment:
As one minor additional note, something here looks 'unsafe' -- will the error response always be here? will it always have a username? Is there no need at all to check that these values `!= null` ?

Chalk my questions up to paranoia 😅

Reply:
Thanks for the raising this. I have double checked, `
createAuthorizationResultWithErrorResponse` will return you a non-null error response, so we are good here.